ops/main.py

```diff
@@ -162,21 +163,38 @@ def main(charm_class):
     """
     charm_dir = _get_charm_dir()
 
+    model_backend = ops.model.ModelBackend()
+    debug = ('JUJU_DEBUG' in os.environ)
+    setup_root_logging(model_backend, debug=debug)
+
     # Process the Juju event relevant to the current hook execution
     # JUJU_HOOK_NAME, JUJU_FUNCTION_NAME, and JUJU_ACTION_NAME are not used
     # in order to support simulation of events from debugging sessions.
     #
     # TODO: For Windows, when symlinks are used, this is not a valid
     #       method of getting an event name (see LP: #1854505).
     juju_exec_path = Path(sys.argv[0])
+    has_dispatch = juju_exec_path.name == 'dispatch'
+    if has_dispatch:
+        # The executable was 'dispatch', which means the actual hook we want to
+        # run needs to be looked up in the JUJU_DISPATCH_PATH env var, where it
+        # will be a path relative to the charm directory (the directory that
+        # holds `dispatch`). If that path actually exists, we want to run that
+        # before continuing.
+        dispatch_path = juju_exec_path.parent / Path(os.environ['JUJU_DISPATCH_PATH'])
```

Comment:
I'm still unsure if we're handling `JUJU_DISPATCH_PATH` correctly if it's absolute.

Reply:
an absolute `JUJU_DISPATCH_PATH` makes no sense, so I don't see how it makes a difference if we add it to `juju_exec_path` before checking...?